packages/smart-contracts/src/contracts/lib/SafeERC20.sol

```diff
@@ -48,4 +48,46 @@ library SafeERC20 {
     /* solium-enable security/no-inline-assembly */
     return result;
   }
+  
+  /**
+    * @dev Deprecated. This function has issues similar to the ones found in
```

Comment:
any reason we're not using `safeIncreaseAllowance`?

Reply:
Yes, we authorize the max allowance. Same as the payment proxies fyi. It's important if you use it to reduce allowance.